### DISCUSSION TITLE
Model hook is not working for me.

### DISCUSSION THREAD

**Discussion by pk4all** [adonisjs/core#2359] [Help] [answered]
This hook not working in my model
```javascript
@beforeDelete()
  public static async decreaseUserFollower (data: FollowUser) {
     await User.query().where({'id':data?.targetId}).decrement('followers',1)
     await User.query().where('id',data?.uid).decrement('following',1)
    
  }
```
Any suggestion for me?

**Comment by RomainLanz** [comment, marked as answer]
Hey @pk4all! 👋 

What do you mean by "it is not working"?

- Does it not run?
- Does it not do what you expect it to do?

**Reply by pk4all** [reply]
@RomainLanz 

My users table has been not update.
it's not do as we expect it to do...

**Reply by RomainLanz** [reply]
Can you please share the line that is deleting your model?

**Reply by pk4all** [reply]
@RomainLanz 

My delete model code is
```javascript 
var exit=await FollowUser.query().where({uid:uid,targetId:data.targetId}).delete()
```

**Reply by RomainLanz** [reply]
When using the query builder directly to update or delete a model, no hooks will be fired.

> Again, in order for hooks to work, Lucid needs the instance of the model first. If you decide to use the query builder directly, then no hooks will be fired.
> ```js
> await User.query().where('is_verified', false).delete()
> ```

📚 https://preview.adonisjs.com/guides/models/crud-operations#delete
